wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt

```diff
@@ -524,6 +528,113 @@ class KotlinGenerator private constructor(
         .build()
   }
 
+  private fun addDefaultFields(
+    type: MessageType,
+    companionBuilder: TypeSpec.Builder,
+    nameAllocator: NameAllocator
+  ) {
+    type.fieldsAndOneOfFields().filter { it.default != null }.forEach { field ->
+      val fieldName = "DEFAULT_" + nameAllocator[field].toUpperCase(Locale.US)
+      val fieldType = field.getClass().copy(nullable = false)
+      val fieldValue = fieldInitializer(type, field.type(), nameAllocator, field.default)
+      companionBuilder.addProperty(PropertySpec.builder(fieldName, fieldType)
+          .apply {
+            if (field.type().isScalar && field.type() != ProtoType.BYTES) {
+              addModifiers(KModifier.CONST)
+            } else {
+              jvmField()
+            }
+          }
+          .initializer(fieldValue)
+          .build())
+    }
+  }
+
+  private fun fieldInitializer(
+    messageType: MessageType,
+    protoType: ProtoType,
+    nameAllocator: NameAllocator,
+    value: Any?
+  ): CodeBlock {
+    val typeName = protoType.typeName
+    return when {
+      value is List<*> -> buildCodeBlock {
+        add("listOf(")
+        var first = true
```

Comment:
Can't you do a map of each line with a `joinAsString` kind of thing?

Reply:
Good point, I'll try

Reply:
I've actually removed that block of code and simplified the method: https://github.com/square/wire/pull/1023/commits/638226470d90a5feb12215f317e49c67950856f9